### ISSUE TITLE
FileRead command

### ISSUE THREAD

**Issue by BorderCloud** [SeleniumHQ/selenium-ide#854]
## 🚀 Feature Proposal

New command to import the text of a file in a textarea (in a wiki, for example).

## Motivation

No command for the moment and interface of selenium IDE doesn't support a long text with special characters.

## Example
I imagine:

Command1 : FileRead
target: return ${localpath} + "/myfile.txt"
value: varText

Command2 : TYPE
target: id=idtextarea1
value: ${varText}

**Comment by tcwaily** [comment]
Selenium ide can't read the file system, so I guess it should be unsupported

**Comment by tourdedave** [comment]
@tcwaily is right, as a browser extension we don't have direct access to the file system. We'd have to build a file upload mechanism. The simplest thing right now is to do it through a plugin. There's one out there for doing something like this with CSVs ([link](https://github.com/tourdedave/selenium-ide-plugins)). Doing the same thing but for flat text files is an interesting addition to that project. Ultimately, once we move to a native app this will be easier to accomplish.

**Comment by corevo** [comment]
@tourdedave I agree, in the meantime this could be a plugin, once we move to a native app this can definitely be standard library.

**Comment by BorderCloud** [comment]
Almost a year... I have may be an idea.

A quick solution can be to create a simple script 'importFileSe' with 3 parameters:
1. origin file with all tests (Tests.side)
2. root folder of files to import in the final Json
3. output filename (for example FinalTests.side)

Everywhere, where there is the command 'file read'. The script  'importFileSe'  will import the text in the Json and replace 'file read' by 'store'.

Without waiting a native app, you can insert the command 'file read' with the same code than 'store'. If the native app is developped, you can replace the code.

What do you think ?

**Comment by corevo** [comment]
I don't understand, you mean bundle the file inside the json?

**Comment by BorderCloud** [comment]
I want to insert simply the texts of files directly in the Json.

**Comment by BorderCloud** [comment]
More simple again... Everywhere in the JSON, a program will replace a pattern, for example <myData.txt>, by the text in the file.
It works without changing SE IDE...

**Comment by corevo** [comment]
Sure then insert them into the JSON, that's a fine workaround.

**Comment by doanva84** [comment]
SE IDE supports partially reading file, i.e File upload via `<input type = "file"...`
Reference: [#1097](https://github.com/SeleniumHQ/selenium-ide/issues/1097#issuecomment-702658882)

**Comment by BorderCloud** [comment]
For info, while waiting for the readFile command, I wrote a little script in [this project ](https://github.com/BorderCloud/selenium-ide-tool-importFile)

`./importFile.sh input.side -o output.side`
